### PR TITLE
[Feature] Android unity message

### PR DESCRIPTION
**Pull request by dengjeffrey**
+ Adds a container class for Unity Messages that serializes content for Unity to consume

### CODE REVIEW

File: android/shopify_buy_plugin/src/test/java/com/shopify/unity/buy/UnityMessageTest.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.Map;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+
+public class UnityMessageTest {
+
+    @Test
+    public void test_serialize() {
```
Comment (sleroux) [inline android/shopify_buy_plugin/src/test/java/com/shopify/unity/buy/UnityMessageTest.java]: Nit: Doesn't really matter but we can follow captial casing as a convention for naming test cases. Just going based off what we've done for the Mobile Buy SDK https://github.com/Shopify/mobile-buy-sdk-android/blob/master/MobileBuy/buy3-pay-support/src/androidTest/java/com/shopify/buy3/pay/PayCartTest.java

File: android/shopify_buy_plugin/build.gradle [hunk 2]
```diff
@@ -28,6 +28,7 @@ android {
 
 dependencies {
     compile fileTree(dir: 'libs', include: ['*.jar'])
+    compile group: 'com.google.code.gson', name: 'gson', version: '2.3.1'
```
Comment (sleroux) [inline android/shopify_buy_plugin/build.gradle]: Any reason in particular you're pulling in Gson? Android has a `JSONObject` class that you can use to do simple JSON serializing/deserializing.

File: android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/UnityMessage.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package com.shopify.unity.buy;
+
+import com.google.gson.Gson;
+import java.util.UUID;
+
+public class UnityMessage implements JsonSerializable {
+
+    // Using capital case here, so when it is serialized the keys are capitalized
+    public String Content;
```
Comment (sleroux) [inline android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/UnityMessage.java]: Related to the checkstyle errors, we can probably make these lowercased and set the JSON object keys in the `toJsonString` function to be captialized - similar to how we've done in Unity.

File: android/shopify_buy_plugin/build.gradle (hunk 2 above)
Comment (sleroux) [inline android/shopify_buy_plugin/build.gradle]: Hmm I just realized that I don't think JSONObject is mocked when running unit tests.

**Comment by dengjeffrey** [comment]
I made the requested changes, please 👀  again @sleroux

**Comment by dengjeffrey** [comment]
Since we've discussed possibly using https://github.com/square/moshi instead of `gson` and JSONObject. I prefer that we come back and refactor later if we do choose to use one of those, since it's a minute implementation detail

File: android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/UnityMessage.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
+package com.shopify.unity.buy;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.UUID;
+
+public class UnityMessage implements JsonSerializable {
```
Comment (sav007) [inline android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/UnityMessage.java]: make it final to not accidentally extend it

File: android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/UnityMessage.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+    public static String CONTENT_KEY = "Content";
```
Comment (sav007) [inline android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/UnityMessage.java]: final

File: android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/UnityMessage.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    public static String IDENTIFIER_KEY = "Identifier";
+
+    public String content;
```
Comment (sav007) [inline android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/UnityMessage.java]: I think you can make it immutable as they are initialized only in constructor (make them final)

File: android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/UnityMessage.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    private String identifier;
```
Comment (sav007) [inline android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/UnityMessage.java]: should it be public as well?

Comment (dengjeffrey) [inline android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/UnityMessage.java]: No, this is a property that won't be used by any interface on Android

**Pull request merged**